lib/Target/TfheRustHL/TfheRustHLEmitter.cpp

```diff
@@ -383,6 +388,15 @@ LogicalResult TfheRustHLEmitter::printBinaryOp(::mlir::Value result,
                                                std::string_view op) {
   emitAssignPrefix(result);
 
+  if (auto cteOp = dyn_cast<mlir::arith::ConstantOp>(rhs.getDefiningOp())) {
+    auto intValue =
+        cast<IntegerAttr>(cteOp.getValue()).getValue().getZExtValue();
+    os << checkOrigin(lhs) << variableNames->getNameForValue(lhs) << " " << op
+       << " " << intValue << "u" << cteOp.getType().getIntOrFloatBitWidth()
+       << ";\n";
```

Comment:
```suggestion
    // Note: arith.constant op requires signless integer types, but here we manually emit an unsigned integer type.
    os << checkOrigin(lhs) << variableNames->getNameForValue(lhs) << " " << op
       << " " << intValue << "u" << cteOp.getType().getIntOrFloatBitWidth()
       << ";\n";
```

since that's why we can't use convertType, I think!